tables/detailed_billing_report/detailed_billing_report_table.go

```diff
@@ -0,0 +1,86 @@
+package detailed_billing_report
+
+import (
+	"strings"
+	"time"
+
+	"github.com/rs/xid"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-aws/sources/s3_bucket"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const DetailedBillingReportTableIdentifier = "aws_detailed_billing_report"
+
+type DetailedBillingReportTable struct{}
+
+func (t *DetailedBillingReportTable) Identifier() string {
+	return DetailedBillingReportTableIdentifier
+}
+
+func (t *DetailedBillingReportTable) GetSourceMetadata() []*table.SourceMetadata[*DetailedBillingReport] {
+	defaultS3ArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
+		FileLayout: utils.ToStringPointer("%{DATA:prefix}/%{DATA:exportName}/%{DATA:folderName}/%{DATA:billing_period}/%{DATA:assembly_id}/%{DATA}.csv.(?:gz|zip)"),
+	}
+
+	return []*table.SourceMetadata[*DetailedBillingReport]{
+		{
+			SourceName: s3_bucket.AwsS3BucketSourceIdentifier,
+			Options: []row_source.RowSourceOption{
+				artifact_source.WithDefaultArtifactSourceConfig(defaultS3ArtifactConfig),
+				artifact_source.WithArtifactExtractor(NewDetailedBillingReportExtractor()),
+			},
+		},
+		{
+			SourceName: constants.ArtifactSourceIdentifier,
+			Options: []row_source.RowSourceOption{
+				artifact_source.WithArtifactExtractor(NewDetailedBillingReportExtractor()),
+			},
+		},
+	}
+}
+
+func (t *DetailedBillingReportTable) EnrichRow(row *DetailedBillingReport, sourceEnrichmentFields schema.SourceEnrichment) (*DetailedBillingReport, error) {
+	row.CommonFields = sourceEnrichmentFields.CommonFields
+	row.TpID = xid.New().String()
+	row.TpIngestTimestamp = time.Now()
+
+	// Fallbacks for TpIndex and TpTimestamp
+	if typehelpers.SafeString(row.LinkedAccountId) != "" {
+		row.TpIndex = *row.LinkedAccountId
+	} else if typehelpers.SafeString(row.PayerAccountId) != "" {
+		row.TpIndex = *row.PayerAccountId
+	} else {
+		row.TpIndex = schema.DefaultIndex
+	}
```

Comment:
```suggestion
```

I think we should not use a default index, since this would only happen if a user modifies their billing report to remove the account IDs above. I think Tailpipe should error on those rows since `tp_index` is missing instead so the users can analyze/fix their data.

Reply:
Done!